batching/batching.go

```diff
@@ -115,19 +116,24 @@ func (b Batches) Iter(ctx context.Context) <-chan data.Entry {
 func (b Batches) Len() int {
 	l := 0
 	for _, batch := range b {
-		l += len(batch)
+		l += len(batch.Data)
 	}
 	return l
 }
 
+type Batch struct {
+	Data
```

Comment:
You embedding Data here, but you are referencing Data directly everywhere.  You don't get any advantages by doing this as Data doesn't have any methods you are promoting and you don't get free index referencing my embedding a map type.  So you should make this `Data Data`.